src/Storages/RabbitMQ/StorageRabbitMQ.cpp

```diff
@@ -393,6 +676,53 @@ bool StorageRabbitMQ::streamToViews()
     std::atomic<bool> stub = {false};
     copyData(*in, *block_io.out, &stub);
 
+    /* Need to stop loop even if connection is ok, because sending ack() with loop running in another thread will lead to a lot of data
```

Comment:
Maybe we can have a separate connection for it? Current logic is quite complicated, needs additional flags to stop the loop, to deactivate tasks, and so on.

Reply:
Can't have a separate connection, because: there is only one unique event loop for each connection, loop handles onReceived() callbacks for channel, which made consume() call, therefore committing messages with channel->ack() needs to be called on the same channel, this means that the channel and the loop must share the same connection as a loop declared with a different connection will not be able to access those channel's callbacks